terraform/function/pubsub/main.tf

```diff
@@ -0,0 +1,41 @@
+resource "google_storage_bucket_object" "functioncode" {
+  name   = "function_sources/${var.function_name}${var.branch_suffix}/source.zip"
+  bucket = var.bucket_name
+  source = var.path_to_zip_file
+}
+
+resource "google_cloudfunctions_function" "function" {
```

Comment:
Can we name this "event_function" or "pubsub_function"? To make te state more readable